scripts/commands/checkLinks.ts

```diff
@@ -122,69 +43,131 @@ function markdownFromNotebook(source: string): string {
   return markdown;
 }
 
-async function checkLinksInFile(
-  filePath: string,
-  existingPaths: string[],
-): Promise<boolean> {
-  if (
-    filePath.startsWith("docs/api/qiskit") ||
-    filePath.startsWith("docs/api/qiskit-ibm-provider") ||
-    filePath.startsWith("docs/api/qiskit-ibm-runtime")
-  ) {
-    return true;
+/**
+ * Return a list of File objects with all the files
+ * in `filePaths` and a list of Link objects with all
+ * the links found in those files.
+ */
+async function loadFilesAndLinks(
+  filePaths: string[],
+): Promise<[File[], Link[]]> {
+  const fileList: File[] = [];
+  const linkList: Link[] = [];
+
+  // Auxiliary Map to avoid link duplications
+  const linkMap = new Map<string, string[]>();
+
+  for (let filePath of filePaths) {
+    const source = await readFile(filePath, { encoding: "utf8" });
```

Comment:
Using `await` in a for loop like this is usually less efficient than you want because it doesn't allow you to leverage your computer's multiprocessing. You can use `Promise.all()` to evaluate a collection/array of `Promise`s so they run in concurrency. That will make this script faster.

I'd recommend creating an `async function` that operates on a single file. It gets all the information you need for this `loadFilesAndLinks`. Then, you can call `await Promise.all(filePaths.map((fp) => loadFile(fp))`